packages/dds/task-manager/src/test/taskManager.spec.ts

```diff
@@ -280,6 +281,99 @@ describe("TaskManager", () => {
+                assert.ok(!taskManager2.assigned(taskId), "Task manager 2 should not be assigned");
+
+                assert.throws(() => { taskManager2.complete(taskId); }, "Should throw error");
+                containerRuntimeFactory.processAllMessages();
+                assert.ok(taskManager1.assigned(taskId), "Task manager 1 should be assigned");
+            });
+
+            it("Can complete a task and remove other clients from queue", async () => {
+                const taskId = "taskId";
+                const volunteerTaskP1 = taskManager1.volunteerForTask(taskId);
+                const volunteerTaskP2 = taskManager2.volunteerForTask(taskId);
+
+                containerRuntimeFactory.processAllMessages();
+                const isAssigned1 = await volunteerTaskP1;
+                assert.ok(isAssigned1, "Should resolve true");
+
+                assert.ok(taskManager1.assigned(taskId), "Task manager 1 should be assigned");
+                assert.ok(taskManager2.queued(taskId), "Task manager 2 should be queued");
+                assert.ok(!taskManager2.assigned(taskId), "Task manager 2 should not be assigned");
+
+                taskManager1.complete(taskId);
+                containerRuntimeFactory.processAllMessages();
+                const isAssigned2 = await volunteerTaskP2;
+                assert.ok(!isAssigned2, "Should resolve false");
+
+                assert.ok(!taskManager1.queued(taskId), "Task manager 1 should not be queued");
+                assert.ok(!taskManager2.queued(taskId), "Task manager 2 should not be queued");
+            });
+
+            it("Can complete a task and remove other subscribed clients from queue", async () => {
+                const taskId = "taskId";
+                taskManager1.subscribeToTask(taskId);
+                taskManager2.subscribeToTask(taskId);
+                containerRuntimeFactory.processAllMessages();
+
+                assert.ok(taskManager1.assigned(taskId), "Task manager 1 should be assigned");
+                assert.ok(taskManager2.queued(taskId), "Task manager 2 should be queued");
+                assert.ok(!taskManager2.assigned(taskId), "Task manager 2 should not be assigned");
+
+                taskManager1.complete(taskId);
+                containerRuntimeFactory.processAllMessages();
+
+                assert.ok(!taskManager1.queued(taskId), "Task manager 1 should not be queued");
+                assert.ok(!taskManager2.queued(taskId), "Task manager 2 should not be queued");
+            });
+
+            it("Can emit completed event", async () => {
+                const taskId = "taskId";
+                const volunteerTaskP1 = taskManager1.volunteerForTask(taskId);
+                taskManager2.subscribeToTask(taskId);
+
+                containerRuntimeFactory.processAllMessages();
+                await volunteerTaskP1;
+
+                const timeoutMs = 100;
+                await timeoutPromise((resolve) => {
```

Comment:
Probably don't need a timeoutPromise here?  In the course of the final `processAllMessages()` the event should fire synchronously, so can probably do something similar to this:
https://github.com/microsoft/FluidFramework/blob/main/packages/dds/task-manager/src/test/taskManager.spec.ts#L266

Also I'm now realizing I've overlooked adding eventing tests in a number of places :-P